### PR TITLE
Progress RVL/MTX

### PR DESCRIPTION
**Pull request by wait-wtf**
- [x] PSMTXIdentity
- [x] PSMTXCopy
- [x] PSMTXConcat
- [x] PSMTXConcatArray
- [x] PSMTXInverse
- [x] PSMTXInvXpose
- [x] PSMTXRotRad
- [x] PSMTXRotTrig
- [x] __PSMTXRotAxisRadInternal
- [x] PSMTXRotAxisRad
- [x] PSMTXTrans
- [x] PSMTXTransApply
- [x] PSMTXScale
- [x] PSMTXScaleApply
- [x] PSMTXQuat
- [x] C_MTXLookAt
- [x] C_MTXLightFrustum
- [x] C_MTXLightPerspective
- [x] C_MTXLightOrtho
- [x] PSMTXMultVec
- [x] PSMTXMultVecSR
- [x] C_MTXFrustum
- [x] C_MTXPerspective
- [x] C_MTXOrtho
- [x] PSVECAdd
- [x] PSVECScale
- [x] PSVECNormalize
- [x] PSVECMag
- [x] PSVECDotProduct
- [x] PSVECCrossProduct
- [x] C_VECHalfAngle
- [x] PSVECSquareDistance
- [x] PSQUATMultiply
- [x] PSQUATScale
- [x] PSQUATDotProduct
- [x] PSQUATNormalize
- [x] PSQUATInverse
- [x] C_QUATMtx
- [x] C_QUATLerp
- [ ] C_QUATSlerp (Got [Error 10342](https://docs.nxp.com/bundle/GUID-EC03B1E9-8136-4FAF-BAF4-98C41299858D/page/GUID-FCE21026-5F30-4176-8F09-69D2AD067507.html))

### CODE REVIEW

**Comment by wait-wtf** [comment]
@riidefi This is ready to merge. I'm having problems with `PSMTXRotRad` though that prevents me from decompiling the rest.

File: source/rvl/mtx/rvlMtx.c [hunk 1]
```diff
@@ -306,3 +306,70 @@ asm u32 PSMTXInvXpose(const register Mtx src, register Mtx invX) {
...
+}
+
+asm void PSMTXRotTrig(register Mtx m, register char axis, register f32 sinA,
+                      register f32 cosA) {
+  nofralloc;
+  frsp fp5, fp1;
+  ori r0, r4, 0x20;
+  frsp fp4, fp2;
+  cmplwi r0, 0x78;
+  lfs fp0, -0x672c(r2);
```
Comment (riidefi) [inline source/rvl/mtx/rvlMtx.c:329]: Directly using r2 offsets like this isn't going to be shiftable, definitely make a note here.

Comment (wait-wtf) [reply source/rvl/mtx/rvlMtx.c:329]: Good find, I forgot about that. I had particular pain generating the "prologue" of this function from C, which is why I had to move it all to asm. I'll see if there's some trick to reference sdata2 properly from asm. I know it's possible for sdata.

Comment (wait-wtf) [reply source/rvl/mtx/rvlMtx.c]: Fixed.

**Pull request merged**